src/EditorFeatures/CSharp/StringCopyPaste/KnownSourcePasteProcessor.cs

```diff
@@ -132,38 +127,169 @@ private void AppendContentForLiteral(StringBuilder builder)
+
+            // A newline and the indentation to start with.  Note: adding the indentation hear means that existing
+            // content will start at the right location, as will any content we are pasting in.
+            if (convertToMultiLine)
+                edits.Add(new TextChange(new TextSpan(StringExpressionBeforePasteInfo.StartDelimiterSpan.End, 0), NewLine + indentationWhitespace));
+
+            // If we need to add braces to existing interpolations, do so now for the interpolations before the selection.
+            if (dollarSignsToAdd != null)
+                UpdateExistingInterpolationBraces(edits, beforeSelection: true, dollarSignsToAdd.Length);
+
+            // Now determine the actual content to add again, this time properly emitting it with
+            // indentation/interpolations correctly.
+            edits.Add(GetContentEditForRawString(insertInterpolations: true, finalDollarSignCount, indentationWhitespace));
+
+            // If we need to add braces to existing interpolations, do so now for the interpolations before the selection.
+            if (dollarSignsToAdd != null)
+                UpdateExistingInterpolationBraces(edits, beforeSelection: false, dollarSignsToAdd.Length);
+
+            // A final new-line and indentation before the end delimiter.
+            if (convertToMultiLine)
+                edits.Add(new TextChange(new TextSpan(StringExpressionBeforePasteInfo.EndDelimiterSpan.Start, 0), NewLine + indentationWhitespace));
+
+            // Then  any extra quotes to the end delimiter.
+            if (quotesToAdd != null)
+                edits.Add(new TextChange(new TextSpan(StringExpressionBeforePasteInfo.EndDelimiterSpanWithoutSuffix.End, 0), quotesToAdd));
+
+            return edits.ToImmutable();
+        }
+
+        private void UpdateExistingInterpolationBraces(
+            ArrayBuilder<TextChange> edits, bool beforeSelection, int dollarSignsToAdd)
+        {
+            var interpolatedStringExpression = (InterpolatedStringExpressionSyntax)StringExpressionBeforePaste;
+
+            foreach (var content in interpolatedStringExpression.Contents)
+            {
+                if (content is InterpolationSyntax interpolation)
+                {
+                    if (beforeSelection && interpolation.Span.End > _selectionSpanBeforePaste.Start)
+                        continue;
+
+                    if (!beforeSelection && interpolation.Span.Start < _selectionSpanBeforePaste.End)
+                        continue;
+
+                    edits.Add(new TextChange(new TextSpan(interpolation.OpenBraceToken.Span.End, 0), new string('{', dollarSignsToAdd)));
+                    edits.Add(new TextChange(new TextSpan(interpolation.CloseBraceToken.Span.Start, 0), new string('}', dollarSignsToAdd)));
+                }
+            }
+        }
+
+        private TextChange GetContentEditForRawString(
```

Comment:
ignore the diff in this mehtod.  it comes from the fact that i merged two similar methods into one (talked above above), but then added a method for the raw case.  the diff thinkgs the raw-method shoudl diff against one of the prior methods.